UM/Math/Polygon.py

```diff
@@ -175,8 +175,10 @@ def intersectsPolygon(self, other: "Polygon") -> Optional[Tuple[float, float]]:
 
         polygon_me = ShapelyUtil.polygon2ShapelyPolygon(self)
         polygon_other = ShapelyUtil.polygon2ShapelyPolygon(other)
-
-        if not (polygon_me.is_valid and polygon_other.is_valid):
+        if not polygon_other.is_empty and not polygon_me.is_empty:
+            if not polygon_me.is_valid and polygon_other.is_valid:  # If not valid
```

Comment:
I'm sorry, but Nallath made a mistake here. The brackets should not have been removed in this one.

It should say either:

```suggestion
            if not (polygon_me.is_valid and polygon_other.is_valid):  # If not valid
```

Or:
```suggestion
            if not polygon_me.is_valid or not polygon_other.is_valid:  # If not valid
```